packages/financial-templates-lib/src/logger/SlackTransport.js

```diff
@@ -163,6 +163,71 @@ function slackFormatter(info) {
   }
 }
 
+class SlackHook extends Transport {
+  constructor(opts) {
+    super(opts);
+    opts = opts || {};
+    this.name = opts.name || "slackWebhook";
+    this.level = opts.level || undefined;
+    this.webhookUrl = opts.webhookUrl;
+    this.formatter = opts.formatter || undefined;
+    this.mrkdwn = opts.mrkdwn || false;
+
+    this.axiosInstance = axios.create({
+      proxy: opts.proxy || undefined
+    });
+  }
+
+  async log(info, callback) {
+    let payload = {
+      mrkdwn: this.mrkdwn
+    };
+
+    let layout = this.formatter(info);
+    payload.text = layout.text || undefined;
+    payload.attachments = layout.attachments || undefined;
+    payload.blocks = layout.blocks || undefined;
+    let errorThrown = false;
+    // If the overall payload is less than 3000 chars then we can send it all in one go to the slack API.
+    if (JSON.stringify(payload).length < 3000) {
+      let response = await this.axiosInstance.post(this.webhookUrl, payload);
+      if (response.status != 200) errorThrown = true;
+    } else {
+      // If it's more than 3000 chars then we need to split the message sent to slack API into multiple calls.
+      let messageIndex = 0;
+      let processedBlocks = [[]];
+      for (let block of payload.blocks) {
+        if (JSON.stringify(block).length > 3000) {
+          // If the block (one single part of a message) is larger than 3000 chars then we must redact part of the message.
+          const stringifiedBlock = JSON.stringify(block);
+          const redactedBlock =
+            stringifiedBlock.substr(0, 1400) +
```

Comment:
What's the 1400 magic number?

Reply:
The upper bound for a message is 3000 charts. I set this number to 1400 chars on each end of the truncation to leave ~100 chars free in the block to ensure that it doesn't overflow.